docs/community/liaoliao666-react-query-kit.md

```diff
@@ -0,0 +1,77 @@
+---
+id: liaoliao666-react-query-kit
+title: React Query Kit
+---
+
+🕊️ A toolkit for ReactQuery that make ReactQuery hooks reusable and typesafe
+
+## what do you benefit from it
+
+- Make `queryKey` strongly related with `queryFn`
+- Manage `queryKey` in a type-safe way
+- Generate quickly a custom ReactQuery hook
+- Make `queryClient`'s operations more clearly associated with which custom ReactQuery hook
+- Set defaultOptions for custom ReactQuery hooks easier and clearer
+
+## Installation
+
+This module is distributed via [NPM](https://www.npmjs.com/package/react-query-kit) and
+should be installed as one of your project's `dependencies`:
+
+```bash
+$ npm i react-query-kit
+# or
+$ yarn add react-query-kit
+```
+
+## Quick start with nextjs
+
+[CodeSandbox](https://codesandbox.io/s/example-react-query-kit-nextjs-uldl88)
+
+```ts
+import { QueryClient, dehydrate } from '@tanstack/react-query'
+import { createQuery } from 'react-query-kit'
+
+type Response = { title: string; content: string }
+type Variables = { id: number }
+
+const usePost = createQuery<Response, Variables, Error>({
+  primaryKey: '/posts',
+  queryFn: ({ queryKey: [primaryKey, variables] }) => {
+    // primaryKey equals to '/posts'
+    return fetch(`${primaryKey}/${variables.id}`).then(res => res.json())
+  },
+  suspense: true
+})
+
+const variables = { id: 1 }
+
+export default function Page() {
+  // queryKey equals to ['/posts', { id: 1 }]
+  const { data } = usePost({ variables, suspense: true })
+
+  return (
+    <div>
+      <div>{data?.title}</div>
+      <div>{data?.content}</div>
+    </div>
+  )
+}
+
+console.log(usePost.getKey()) //  ['/posts']
+console.log(usePost.getKey(variables)) //  ['/posts', { id: 1 }]
+
+export async function getStaticProps() {
+  const queryClient = new QueryClient()
+
+  await queryClient.prefetchQuery(usePost.getKey(variables), usePost.queryFn)
```

Comment:
wouldn't `usePost.prefetch()` be nice?

Reply:
I wanna the custom hook is agnostic with which `queryClient`.

Reply:
If the api design as this. it would also get a problem form such api.
```
// Here is type definition
type prefetch = (variables?: Variables, options?:FetchQueryOptions) => Promise<void>

// If the `prefetch` do not need pass `variables`
// but I need pass `options`
// it will be like below. it's ugly
usePost.prefetch(undefined, options)
```
Even If I can do it. `queryClient` also has many apis, do I need to provide all of these api😂.

I don't think it's a wise choice